quinn/src/recv_stream.rs

```diff
@@ -21,8 +21,40 @@ use crate::{
 /// - A variant of [`ReadError`] has been yielded by a read call
 /// - [`stop()`] was called explicitly
 ///
+/// # Closing a stream
+///
+/// When a stream is expected to be closed gracefully the sender should call
+/// [`SendStream::finish`].  However there is no guarantee the connected [`RecvStream`] will
+/// receive this End Of File (EOF) indicator in the same QUIC frame as the last frame which
+/// carried data.
```

Comment:
"EOF" isn't really a QUIC term. Maybe `the "finished" notification`?

Reply:
True, and here I agree because this describes QUIC behaviour.  Later when talking about how to use the stream I would keep the EOF terminology however because it is an `AsyncRead` concept and that is what you're interacting with.

Reply:
> it is an `AsyncRead` concept and that is what you're interacting with.

None of the methods in the example involve `AsyncRead`, nor is `AsyncRead` in any way necessary to use `RecvStream`s. Due to the baked in `io::Error` it's really a second-class API, exposed only for compatibility.

It's true that EOF is usually a familiar concept all the same, but I worry that it will be confusing to have inconsistent terminology.

Reply:
Ok, I've fully removed this.